projects/ion/src/lib/pagination/pagination.component.html

```diff
@@ -23,9 +23,34 @@
     [disabled]="loading || !hasPrevious()"
   ></ion-button>
 
-  <div class="pages-list">
+  <div
+    *ngIf="isAdvanced; else regularPagination"
+    class="pages-list"
+    data-testid="advanced-pagination"
+  >
     <button
-      *ngFor="let page of pages"
+      [class]="'square-pag ' + 'page-' + size"
+      [class.selected]="pages[0].selected"
+      [attr.data-testid]="'page-' + pages[0].page_number"
+      (click)="selectPageOnClick(pages[0].page_number)"
+      [class.disabled]="loading"
+      [disabled]="loading"
+    >
+      <span>{{ pages[0].page_number }}</span>
+    </button>
+    <ion-button
+      [iconType]="hoverControl.left ? 'left3' : 'more'"
+      data-testid="more-left"
+      type="secondary"
+      [size]="size"
+      [disabled]="loading"
+      (mouseenter)="changeIconHover('left', true)"
+      (mouseleave)="changeIconHover('left', false)"
+      (click)="jumpPagesBackward()"
+      *ngIf="currentPage().page_number >= 5"
```

Comment:
Change this validation to a variable, because functions will be getting called a large amount of times.